odbc/src/api/functions.rs

```diff
@@ -1348,8 +1348,46 @@ fn sql_free_handle(handle_type: HandleType, handle: *mut MongoHandle) -> Result<
 ///
 #[named]
 #[no_mangle]
-pub unsafe extern "C" fn SQLFreeStmt(statement_handle: HStmt, _option: SmallInt) -> SqlReturn {
-    unimpl!(statement_handle);
+pub unsafe extern "C" fn SQLFreeStmt(statement_handle: HStmt, option: SmallInt) -> SqlReturn {
+    // unimpl!(statement_handle);
+    panic_safe_exec_clear_diagnostics!(
+        debug,
+        || {
+            let mongo_handle = MongoHandleRef::from(statement_handle);
+            let stmt = must_be_valid!((*mongo_handle).as_statement());
+
+            match FromPrimitive::from_i16(option) {
+                // Drop all pending results from the cursor and close the cursor.
+                Some(FreeStmtOption::SQL_CLOSE) => {
+                    let _ = stmt
+                        .mongo_statement
+                        .write()
+                        .unwrap()
+                        .as_mut()
+                        .unwrap()
+                        .close_cursor();
+                    SqlReturn::SUCCESS
+                }
+                // Release all column buffers bound by SQLBindCol by removing the bound_cols map.
+                Some(FreeStmtOption::SQL_UNBIND) => {
+                    *stmt.bound_cols.write().unwrap() = None;
+                    SqlReturn::SUCCESS
+                }
+                // We do not implement SQLBindParameter, so this is a no-op.
+                Some(FreeStmtOption::SQL_RESET_PARAMS) => SqlReturn::SUCCESS,
+                _ => {
+                    // TODO: This is handled by the DM, so really this is an unreachable branch.
+                    //       Should we include the error in diagnostics or is it not worth?
+                    // add_diag_info!(
+                    //     mongo_handle,
+                    //     ODBCError::InvalidAttrIdentifier(option as i32)
+                    // );
```

Comment:
Thoughts on this? Jonathan suggested we _may_ be ok to not include this error here since this is not a case we should ever hit since the DM should ensure the function call is rerouted if an invalid option comes in. I'm happy to remove it but figured I'd put it into review for confirmation.

Reply:
looking at the code now I think I agreew with what I said earlier, I dont know if we should do additional work to ultimately return an error. my mental rust compiler is a little weak but could we put unreachable here with a coment? or does it have to be an error to satisfy the return type.

We should, however, probably handle `None` separately then the default error here. I think we need to return `HY092` as the state with the message `Option type out of range` if we get a value that isn't one of these 4. We'll likely need to create a new error for this

Reply:
Oh wait I totally didn't notice the "(DM)" next to the "Option type out of range" error on the docs page before. I thought that the only driver manager errors were the "Function sequence" and "Driver does not support this function" errors. Given that, do we need any actual true error handling here? Could/should we just do a wildcard for everything other than the 3 real options, and have that wildcard be either `unreachable!()` or `SqlReturn::ERROR`?

Reply:
ah good catch. In that case I think we might be good then to just wildcard it (sorry for the misdirect). I feel like `SqlReturn::ERROR` might be more consistent with what we do in other places in the driver but I could see it going either way!